spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/sql/CloudSqlEnvironmentPostProcessor.java

```diff
@@ -80,11 +80,15 @@ public void postProcessEnvironment(
           .getPropertySources()
           .addFirst(new MapPropertySource("CLOUD_SQL_DATA_SOURCE_URL", primaryMap));
 
-      CredentialsPropertiesSetter.setCredentials(sqlProperties, propertiesRetriever.getGcpProperties());
+      CredentialsPropertiesSetter.setCredentials(sqlProperties,
+          propertiesRetriever.getGcpProperties());
 
       // support usage metrics
-      CoreSocketFactory.setApplicationName(
-          "spring-cloud-gcp-sql/" + this.getClass().getPackage().getImplementationVersion());
+      String applicationName =
+          "spring-cloud-gcp-sql/" + this.getClass().getPackage().getImplementationVersion();
+      if (!CoreSocketFactory.getApplicationName().contains(applicationName)) {
```

Comment:
Hi. I'm just wondering if this kind of logic actually belongs inside of `CoreSocketFactory::setApplicationName`. Just a thought ...